### DISCUSSION TITLE
Vanilla html/js bug since 2.7.0 when initiating Web3Modal()

### DISCUSSION THREAD

**Discussion by ismoli** [WalletConnect/walletconnect-monorepo#3306] [Developer Support] [answered]
Found a bug since the version 2.7.0 (working fine in 2.6.2):

when using vanilla html/js and following example https://github.com/WalletConnect/web3modal-examples/tree/main/web3modal-wagmi-html-cdn.
Getting the following error in console upon initiating Web3Modal (export const web3Modal = new Web3Modal()....) :
`
hey-listen.es.js:3 Uncaught (in promise) ReferenceError: process is not defined
    at hey-listen.es.js:3:5
(anonymous) @ hey-listen.es.js:3
await in (anonymous) (async)
_e @ client.ts:25
(anonymous) @ main.js:43
`
following line in hey-listen.js 

`if (process.env.NODE_ENV !== 'production') {`

This obviously breaks whole thing and 'Connect' button is not rendering

**Comment by glitch-txs** [comment, marked as answer]
Thanks for reporting, at the meantime you can add this polyfill at the top of your code
```ts
  if (!window.process) {
    window.process = { env: {} }
  }
```
